### PR TITLE
chore: PoC - Model generation - Supporting nested schema (objects - Map, MapNested, SingleNested Attributes)

### PR DESCRIPTION
**Pull request by maastha**
## Description

PoC - Model generation - Supporting nested schema (objects - Map, MapNested, SingleNested Attributes)

Link to any related issue(s): [CLOUDP-278226](https://jira.mongodb.org/browse/CLOUDP-278226)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: tools/codegen/codespec/api_to_provider_spec_mapper_test.go [hunk 1]
```diff
@@ -129,12 +129,6 @@ func TestConvertToProviderSpec_nested(t *testing.T) {
 								String:                   &codespec.StringAttribute{},
 								Description:              conversion.StringPtr(testPathParamDesc),
 							},
-							{
```
Comment (maastha) [inline tools/codegen/codespec/api_to_provider_spec_mapper_test.go:132]: cleaning up some attrs so the test focuses on nested attrs only

File: tools/codegen/codespec/api_to_provider_spec_mapper_test.go [hunk 2]
```diff
@@ -156,9 +150,43 @@ func TestConvertToProviderSpec_nested(t *testing.T) {
...
+												Name:                     "attr",
+												ComputedOptionalRequired: codespec.Optional,
 												String:                   &codespec.StringAttribute{},
+											},
+										},
+									},
+								},
+							},
+							{
+								Name:                     "nested_set_array_attr",
```
Comment (maastha) [inline tools/codegen/codespec/api_to_provider_spec_mapper_test.go:181]: added nested set attr

File: tools/codegen/codespec/attribute.go [hunk 3]
```diff
@@ -46,7 +46,10 @@ func (s *APISpecSchema) buildResourceAttr(name string, computability ComputedOpt
 	case OASTypeArray:
 		return s.buildArrayAttr(name, computability)
 	case OASTypeObject:
-		return nil, nil // TODO: add support for SingleNestedObject and MapAttribute
+		if s.Schema.AdditionalProperties != nil && s.Schema.AdditionalProperties.IsA() {
```
Comment (AgustinBettati) [inline tools/codegen/codespec/attribute.go:49]: q: what does `IsA()` imply?

File: tools/codegen/codespec/testdata/api-spec.yml [hunk 4]
```diff
@@ -376,20 +404,30 @@ components:
...
+        - nestedListArrayAttr
+    SimpleStringRefObject:
+      type: string
+      description: Test field description
     NoBody:
       type: object
       description: Endpoint does not return a response body.
+    NestedMapObjectAttr:
+      type: object
+      additionalProperties:
```
Comment (AgustinBettati) [inline tools/codegen/codespec/testdata/api-spec.yml:428]: q: curious if in our api spec there is use of `additionalProperties` for representing generic maps.

**Review by AgustinBettati — APPROVED** [review]
LGTM

File: tools/codegen/codespec/testdata/api-spec.yml (hunk 4 above)
Comment (maastha) [reply tools/codegen/codespec/testdata/api-spec.yml:428]: There is. Connection strings is one example https://github.com/mongodb/atlas-sdk-go/blob/main/openapi/atlas-api-transformed.yaml#L22082

File: tools/codegen/codespec/attribute.go (hunk 3 above)
Comment (maastha) [reply tools/codegen/codespec/attribute.go:49]: libopenapi stores additionalProperties as Dynamic values (`AdditionalProperties *DynamicValue[*SchemaProxy, bool] `) defined here https://github.com/pb33f/libopenapi/blob/eff4ce83e7814228bc50e34e5e60c4312a041e98/datamodel/high/base/dynamic_value.go#L20 "The left value (A) is a 3.0 schema property value"

File: tools/codegen/codespec/testdata/api-spec.yml (hunk 4 above)
Comment (AgustinBettati) [reply tools/codegen/codespec/testdata/api-spec.yml:428]: great 👍

**Pull request merged**